src/main/kotlin/com/exactpro/th2/codec/DecodeProcessor.kt

```diff
@@ -40,7 +41,17 @@ class DecodeProcessor(
             val data: ByteArray = source.body.toByteArray()
             logger.debug { "Start decoding message with id: '${source.metadata.id.toDebugString()}'" }
             logger.trace { "Decoding message: ${source.toDebugString()}" }
-            val decodedMessages = getCodec().decode(data, source.toCodecContext())
+            val decodedMsg = getCodec().decode(data, source.toCodecContext())
+
+            val decodedMessages = arrayListOf<IMessage>()
+            decodedMsg.forEach { msg ->
+                if (msg.name == EvolutionBatch.MESSAGE_NAME) {
+                    decodedMessages += EvolutionBatch(msg).batch
+                } else {
+                    decodedMessages += msg
+                }
+            }
```

Comment:
Can be replaced with the following code
```suggestion
            val decodedMessages = getCodec().decode(data, source.toCodecContext())
                            .flatMap {
                              if (msg.name == EvolutionBatch.MESSAGE_NAME) {
                                  EvolutionBatch(msg).batch
                              } else {
                                  listOf(msg)
                              }
                            }
```